### PR TITLE
Update for Android Kotlin DSL based projects

### PR DESCRIPTION
**Pull request by theshivamlko**
## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### Changes
Below settings were tested in Kotlin DSL based Android Projects

1. ``` repositories {
        maven(url = "https://storage.googleapis.com/download.flutter.io")
        maven(url = "/Users/shivam/StudioProjects/flutter_module/build/host/outputs/repo")
    }```
 need to be added in `settings.gradle.kts`

ADDED
2. Implementing `profileImplementation` in Kotlin DSL projects

~~**Note**: I still cannot find about `profileImplementation` in DSL. will find and update soon.~~

### CODE REVIEW

**Comment by atsansone** [comment]
This is blocked until https://github.com/flutter/website/pull/9297 merges.

**Comment by atsansone** [comment]
@vashworth : Could you validate the gradle changes added here?

**Comment by vashworth** [comment]
> @vashworth : Could you validate the gradle changes added here?

My gradle knowledge is pretty limited. Perhaps @reidbaker or someone on his team can review

File: src/add-to-app/android/project-setup.md
Comment (camsim99) [inline src/add-to-app/android/project-setup.md:319]: Trying this out with a freshly created Android app + Flutter module and it doesn't seem to work. Is this correct syntax?

If you haven't already, can you validate these changes with a similar setup and comment the steps you took? That's how most developers would land on this page. It would also help me review this because I'm not an add-to-app expert.

**Comment by theshivamlko** [comment]
@camsim99 I am new to add Flutter Module in Android app. This is how i found out to use module in Native. Website process not working.
Here after `flutter build aar` provides instruction to add generated `aar` to Native android.
Steps for `.kts` project
1. Build Flutter AAR in flutter module project using `flutter build aar`.  Files will be generated at path 
` build/host/outputs/repo/<package-name>`.
2. Using the Module in Android Native app. Create new Android Kotlin DSL `build.gradle.kts` project.
3. Goto `app/build.gradle.kts`, add your flutter module dependencies
```
dependencies {
    debugImplementation("com.example.flutter_module:flutter_debug:1.0")
    releaseImplementation("com.example.flutter_module.host:flutter_release:1.0")
}
```

4. Goto `<android-project>/settings.gradle.kts` , add local path to flutter module build.

```
dependencyResolutionManagement {
    repositories {
        maven(url = "https://storage.googleapis.com/download.flutter.io")
        maven(url = "/Users/shivam/StudioProjects/flutter_module/build/host/outputs/repo")
    }
}
```
5. Modules is added and ready to use
6. Here is my practice example of flutter module, [basic_flutter_module](https://github.com/theshivamlko/basic_flutter_module)
7. Implementation of Example Native app, `ConvertorApp.kt` , `MainActivity.kt`
 [UnitConvertorCompose](https://github.com/theshivamlko/Android-Course-Practice/tree/main/UnitConvertorCompose)

**Comment by theshivamlko** [comment]
@atsansone any update on this ?

**Comment by camsim99** [comment]
Sorry for the delay on this, I'll add this to my queue of things to review this week.

**Comment by camsim99** [comment]
@theshivamlko Thank you so much for the detailed steps! Those worked for me, so let's get this change landed :)

File: src/add-to-app/android/project-setup.md
Comment (camsim99) [inline src/add-to-app/android/project-setup.md:298]: We should probably also change what is currently [line 269](https://github.com/flutter/website/blob/a133bb3be86e4f501eb49f6839b5b35bf77faded/src/add-to-app/android/project-setup.md?plain=1#L269) to indicate that the steps will differ for Java and Kotlin projects.

File: src/add-to-app/android/project-setup.md
Comment (camsim99) [inline src/add-to-app/android/project-setup.md:320]: Can we add the `profileImplementation` here as well? Adding any necessary steps needed to add that implementation will keep this consistent with the instructions for Java based projects.

**Review by camsim99 — APPROVED** [review]
Looks good overall, just left a nit and question about `profileImplementation`.

As an aside, we will likely need to file an issue to also update the instructions for this that get printed to the terminal so that it at least mentions where to look for instructions for Kotlin-based projects. I can file this when this lands.

**Review by camsim99 — CHANGES_REQUESTED** [review]
Oops, meant to request changes before this can be merged!

**Comment by sfshaza2** [comment]
@camsim99, thanks so much for your thorough review! @theshivamlko, @camsim99 is a software engineer and not a writer or a DevRel engineer, so we have to be mindful of her already-taxed time.

**Comment by camsim99** [comment]
This should be ready to land when the comments I left are addressed! If you need me to hop in to get this landed, just let me know.

File: src/add-to-app/android/project-setup.md
Comment (theshivamlko) [reply src/add-to-app/android/project-setup.md:366]: Steps updated to add profileImplementation in Kotlin DSL based project

File: src/add-to-app/android/project-setup.md
Comment (domesticmouse) [inline src/add-to-app/android/project-setup.md:353]: This does not look right

Comment (theshivamlko) [reply src/add-to-app/android/project-setup.md]: > This does not look right

I tested this on Groovy and Kotlin DSL based projects.

In Groovy Projects `profileImplementation` is available to use.
`profileImplementation("com.example.flutter_module:flutter_profile:1.0")`

In Koltin DSL `profileImplementation` need to be created in Host Project to use profile version. According to resources i was able to find There multiple ways to implement `profileImplementation`. 

1. `add("profileImplementation", "com.example.flutter_module:flutter_profile:1.0")`  OR
2. `"profileImplementation"("com.example.flutter_module:flutter_profile:1.0")`
[Configuring plugins without type-safe accessors](https://docs.gradle.org/current/userguide/kotlin_dsl.html#ex-configuring-plugins-without-type-safe-accessors)

For sample i have created Project to test. `flutter_module` is used in both host projects.
1. Flutter Module [flutter_module](https://github.com/theshivamlko/basic_flutter_module)
3. Kotlin DSL Android Project [UnitConvertorCompose](https://github.com/theshivamlko/Android-Course-Practice/tree/main/UnitConvertorCompose)
4. Groovy Android Project [FlutterAddToAppGroovy](https://github.com/theshivamlko/FlutterAddToAppGroovy)

Comment (domesticmouse) [reply src/add-to-app/android/project-setup.md]: > In Groovy Projects `profileImplementation` is available to use.
> `profileImplementation("com.example.flutter_module:flutter_profile:1.0")`

That's not what is in the PR. Please re-read the delta.

Comment (camsim99) [reply src/add-to-app/android/project-setup.md]: @bartekpacia can you weigh in here? Not sure what's expected in terms of having a `profileImplementation` for Kotlin DSL based projects.

Comment (domesticmouse) [reply src/add-to-app/android/project-setup.md]: The line I highlighted is invalid syntax IIUC

Comment (bartekpacia) [reply src/add-to-app/android/project-setup.md]: This strange syntax indeed looks valid ([see here](https://docs.gradle.org/current/userguide/kotlin_dsl.html#artifact_configurations)). I've never built an add-to-app project though and am not familiar with the build system there.

File: src/add-to-app/android/project-setup.md
Comment (bartekpacia) [inline src/add-to-app/android/project-setup.md:349]: this is a nit but let's use double-quotes in Groovy as well

File: src/add-to-app/android/project-setup.md
Comment (sfshaza2) [inline src/add-to-app/android/project-setup.md:330]:
```suggestion
After an `aar` build of a Kotlin DSL-based Android project,
follow these steps to add the flutter_module.
```

File: src/add-to-app/android/project-setup.md
Comment (sfshaza2) [inline src/add-to-app/android/project-setup.md:332]:
```suggestion
Include the flutter module as a dependency in 
the Android project's `app/build.gradle` file.
```

File: src/add-to-app/android/project-setup.md
Comment (sfshaza2) [inline src/add-to-app/android/project-setup.md:355]:
```suggestion
The `profileImplementation` ID is a custom `configuration` to be
implemented in the `app/build.gradle` file of a host project.
```

File: src/add-to-app/android/project-setup.md
Comment (sfshaza2) [reply src/add-to-app/android/project-setup.md:362]: It looks like we can resolve this, yes?

File: src/add-to-app/android/project-setup.md
Comment (sfshaza2) [inline src/add-to-app/android/project-setup.md:349]: Following @bartekpacia's suggestion:

```suggestion
  debugImplementation "com.example.flutter_module:flutter_debug:1.0"
```

Comment (sfshaza2) [reply src/add-to-app/android/project-setup.md:349]: I added a suggestion for this.

**Review by sfshaza2 — COMMENTED** [review]
OK! This P4 has been languishing awhile! I've read through all the feedback. It looks like most of @camsim99's changes were implemented, except for one. Camille asked that, at line 269, a note is added saying that the setup for Java and Kotlin are different:

`
We should probably also change what is currently [line 269](https://github.com/flutter/website/blob/a133bb3be86e4f501eb49f6839b5b35bf77faded/src/add-to-app/android/project-setup.md?plain=1#L269) to indicate that the steps will differ for Java and Kotlin projects.
`

Otherwise, after my feedback is incorporated, I think we are ready to land!

**Review by sfshaza2 — APPROVED** [review]
@camsim99, it appears that your comments are addressed? (Or, is it good enough?) My "text writer-y" feedback is addressed. As you requested changes on this PR, we need your explicit approval.  thx!

**Comment by sfshaza2** [comment]
@theshivamlko, I want to thank you for hanging in there! Hopefully we can land this in the next day or two.

**Comment by sfshaza2** [comment]
WOOT! Landing!

**Pull request merged**
